File: Auth-service/src/main/java/com/slavacom/auth_service/notification/model/NotificationRequest.java
package com.slavacom.auth_service.notification.model;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class NotificationRequest {
    private UUID userId;

    @Email(message = "Email должен быть корректным")
    private String email;

    private Long telegramId;

    @NotBlank(message = "Заголовок не должен быть пустым")
    private String title;

    @NotBlank(message = "Сообщение не должно быть пустым")
    private String message;

    @NotEmpty(message = "Должен быть хотя бы один канал")
    private List<Channel> channels;
}

File: OrganizationService/src/main/java/com/slavacom/organizationservice/notification/model/NotificationRequest.kt
package com.slavacom.organizationservice.notification.model

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import java.util.UUID

data class NotificationRequest(
    val userId: UUID? = null,

    @field:Email(message = "Email должен быть корректным")
    val email: String? = null,

    val telegramId: Long? = null,

    @field:NotBlank(message = "Заголовок не должен быть пустым")
    val title: String? = null,

    @field:NotBlank(message = "Сообщение не должно быть пустым")
    val message: String? = null,

    @field:NotEmpty(message = "Должен быть хотя бы один канал")
    val channels: List<Channel>? = null
)
